### PR TITLE
Observing events on the map

### PR DESCRIPTION
**Pull request by Bmaks-github**

### CODE REVIEW

File: src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	/**
+	 * Web service endpoint to fetch all Event points between two coordinates.
+	 * The service returns the list of Point entities as JSON.
+	 * 
+	 * @return A ResponseEntity containing a List of Point objects.
+	 */
+	@RequestMapping(value = "/point", method = RequestMethod.GET,
+			headers = "Accept=application/json", produces = { "application/json" })
+	public ResponseEntity<?> getPlacePointsBetweenTwoCoordinates(
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java]: Plz, rename it to getEventPointsBetweenTwoCoordinates

File: src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		
+		Point southWest = pointService.createPoint(southWestParam);
+		Point northEast = pointService.createPoint(northEastParam);
+		
+		if (southWest.isEmpty() || northEast.isEmpty()) {
+			logger.debug("< Bad Request for getting event points between (" + southWestParam + " - " + northEastParam);
+			return new ResponseEntity<>(points, HttpStatus.BAD_REQUEST);
+		}
+		
+		points = eventService.getEventPointsBetweenTwoCoordinates(southWest, northEast);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java]: Plz, rename it to findEventPointsBetweenTwoCoordinates.
I have done this mistake in PlaceServiceImpl.
I actually have already fixed it for the places in my pull request, that is waiting for the Misha's approve.
PS: notice, that all service classes in our app for reading data from DB use methods find...., not get.... (besides my typos I've mentioned above).

File: src/main/java/ua/softserve/rv_018/greentourism/service/EventService.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+     * Find all Event entities.
+     * @return A List of Event objects.
+     */
+	List<Event> findAll();
+
+	/**
+     * Find all Points entities between two coordinates.
+     * @return A List of Event objects.
+     */
+	List<Point> getEventPointsBetweenTwoCoordinates(Point southWest, Point northEast);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/EventService.java]: Plz, rename it to findEventPointsBetweenTwoCoordinates.
You can find explanation above.

File: src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java (hunk 2 above)
Comment (Bmaks-github) [reply src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java]: Ok.

File: src/main/java/ua/softserve/rv_018/greentourism/model/Event.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	private String name;
+	
+	@ManyToOne(cascade = CascadeType.ALL)
+	@JoinColumn(name = "point_id", nullable = false)
+	private Point point;
+	
+	@ManyToOne(cascade = CascadeType.ALL)
+	@JoinColumn(name = "owner_id")
+	private User user;
+
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Event.java:48]: Plz, provide empty constructor for the class.
I know it works like this for now, but if someone will need to provide not empty constructor, he could be not aware of the fact, that constructor by default will not be provided by the JVM anymore and hibernate couldn't work with this class in this situation.

File: src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+import ua.softserve.rv_018.greentourism.model.Event;
+import ua.softserve.rv_018.greentourism.model.Point;
+import ua.softserve.rv_018.greentourism.service.EventService;
+import ua.softserve.rv_018.greentourism.service.PointService;
+
+@RequestMapping(value = "/api/event")
+@Controller
+public class EventController {
+
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/EventController.java]: Delete this empty line, plz

File: src/main/java/ua/softserve/rv_018/greentourism/repository/EventRepository.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import ua.softserve.rv_018.greentourism.model.Event;
+
+@Repository
+public interface EventRepository extends JpaRepository<Event, Long> {
+
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/repository/EventRepository.java]: Delete this empty line, plz

File: src/main/java/ua/softserve/rv_018/greentourism/model/Event.java (hunk 4 above)
Comment (Bmaks-github) [reply src/main/java/ua/softserve/rv_018/greentourism/model/Event.java:48]: Hmm, strange. Earlier with another model class it didn't work without empty constructor, but now it works even without.

File: src/main/java/ua/softserve/rv_018/greentourism/repository/EventRepository.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import ua.softserve.rv_018.greentourism.model.Event;
+
+@Repository
+public interface EventRepository extends JpaRepository<Event, Long> {
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/repository/EventRepository.java]: Plz, change Long to Integer.
PS: second generic is a type of id field of the first generic, i.e. type of Event.id.
I noticed now, that we have the same problem in the PlaceRepository.

File: src/main/java/ua/softserve/rv_018/greentourism/model/Event.java (hunk 4 above)
Comment (Sabfir) [reply src/main/java/ua/softserve/rv_018/greentourism/model/Event.java:48]: If you don't have any constructor JVM will provide default one (empty) for you. So no having any constructor - it works.
The problem will be when someone add not empty constructor (explanation above, I can give additional explanation during daily).

Comment (Bmaks-github) [reply src/main/java/ua/softserve/rv_018/greentourism/model/Event.java:48]: Ok, thanks.

**Comment by ghost** [comment]
reviewed

**Comment by KostyaTk** [comment]
reviewed

**Pull request merged**